# Clase 6

Mónica hace aretes y cadenitas de joyería... Le toma 30 minutos hacer un par de aretes y una hora hacer una cadenita,
y como Mónica también es estudiahambre, solo dispone de 10 horas a la semana para hacer las joyas.
Por otra parte, el material que compra solo le alcanza para hacer 15 unidades de joyas por semana.

Los aretes dan utilidad de $\$15 $  y $\$20 $ en cadenita. Maximíza su utilidad

<br>

In [1]:
import scipy.optimize as opt
import numpy as np
import sympy as sym
from decimal import *
sym.init_printing(use_latex='mathjax')

La función para maximizar es  $f= 15x_0 + 20x_1$

In [2]:
f = [-15, -20]
f

[-15, -20]

Su matriz es:
<br>
$0.5x_0 + 1x_1 = 10$
<br>
$x_0 + x_1 = 15$

In [3]:
A = [[0.5, 1], [1,1]]
b = [10,15]
A,b

([[0.5, 1], [1, 1]], [10, 15])

In [4]:
x0_bound = (0, None)
x1_bound = (0, None)

In [5]:
res = opt.linprog(f, A_ub=A, b_ub=b, bounds=(x0_bound, x1_bound),options={'disp':True})
res
res.x

Optimization terminated successfully.
         Current function value: -250.000000 
         Iterations: 2


array([10.,  5.])

Nuestro resulatdo nos da que necesita vender **10 aretes y 5 cadenas**, para maximizar su ingreso de **\$250**

***

### Con diferentes parámetros

Mónica hace aretes y cadenitas de joyería... Le toma 15 minutos hacer un par de aretes y 45 min el hacer una cadenita,
y como Mónica también es estudiahambre, solo dispone de 8 horas a la semana para hacer las joyas.
Por otra parte, el material que compra solo le alcanza para hacer 20 unidades de joyas por semana.

Los aretes dan utilidad de $\$15 $  y  $\$20 $ en cadenita. Maximíza su utilidad

La función para maximizar es  $f= 15x_0 + 20x_1$

In [6]:
f = [-15, -20]
f

[-15, -20]

Su matriz es:
<br>
$0.25x_0 + 0.75x_1 = 8$
<br>
$x_0 + x_1 = 20$

In [7]:
A = [[0.25, 0.75], [1,1]]
b = [8,20]
A,b

([[0.25, 0.75], [1, 1]], [8, 20])

In [8]:
x0_bound = (0, None)
x1_bound = (0, None)

In [9]:
res = opt.linprog(f, A_ub=A, b_ub=b, bounds=(x0_bound, x1_bound),options={'disp':True})
res
res.x

Optimization terminated successfully.
         Current function value: -330.000000 
         Iterations: 2


array([14.,  6.])

Nuestro resulatdo nos da que necesita vender **14 aretes y 6 cadenas**, para maximizar un ingreso de **\$330**